dynamodb-stream-eventbridge-fanout/src/test/java/no/unit/nva/doi/publisher/EventBridgeRetryClientTest.java

```diff
@@ -0,0 +1,157 @@
+package no.unit.nva.doi.publisher;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResponse;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResultEntry;
+
+public class EventBridgeRetryClientTest {
+
+    private static final int MAX_ATTEMPT = 3;
+    @Mock
+    private EventBridgeClient eventBridge;
+
+    private EventBridgeRetryClient client;
+
+    /**
+     * Set up environment for test.
+     */
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        client = new EventBridgeRetryClient(eventBridge, MAX_ATTEMPT);
+    }
+
+    @Test
+    public void putEventsHasNoFailure() {
```

Comment:
I think the test is a bit complicated because the testing scenario is complicated. You have one successful and one failing entry and you verify that one passes and the other fails and that for the failing one the called method tries twice.

I would suggest that you break this test in 4 tests:
1. Success when attempting to put the event in the bus returns empty list. (one entry)
2. Failure when attempting to put the event in the bus returns the failed entry. (one entry)
3. Failure when attempting to put the event in the bus means that the method had tried twice (one entry).
4. Partial failure returns only the events that failed to be pushed